File: backend/server-py/app/src/db/schemas/user_words_map.py
from .base import Base, TimestampMixin
import sqlalchemy as sa
from sqlalchemy.dialects import postgresql as pg

class UserWordsMap(Base, TimestampMixin):
    __tablename__ = 'user_words_map'
    id = sa.Column(sa.Integer, primary_key=True)
    user_id = sa.Column(sa.Integer, sa.ForeignKey('users.id'), nullable=False)
    words_id = sa.Column(sa.Integer, sa.ForeignKey('words.id'), nullable=False)
    data = sa.Column(pg.JSONB, nullable=True)  # 비공개 단어 목록
    # data 구조:
    # {
    #   "non_visible_word": ["비공개단어1", "비공개단어2", "비공개단어3"]
    # }

File: backend/server-node/src/db/schemas/user_words_map.js
const { EntitySchema } = require('typeorm');

const UserWordsMapSchema = new EntitySchema({
    name: 'UserWordsMap',
    tableName: 'user_words_map',
    columns: {
        id: {
            primary: true,
            type: 'int',
            generated: true
        },
        user_id: {
            type: 'int',
            nullable: false
        },
        words_id: {
            type: 'int',
            nullable: false
        },
        data: {
            type: 'jsonb',
            nullable: true
        },
        created_at: {
            type: 'timestamp',
            createDate: true,
            nullable: false
        },
        updated_at: {
            type: 'timestamp',
            updateDate: true,
            nullable: false
        }
    },
    relations: {
        user: {
            target: 'User',
            type: 'many-to-one',
            joinColumn: {
                name: 'user_id',
                referencedColumnName: 'id'
            },
            nullable: false
        },
        words: {
            target: 'Words',
            type: 'many-to-one',
            joinColumn: {
                name: 'words_id',
                referencedColumnName: 'id'
            },
            nullable: false
        }
    }
});

module.exports = UserWordsMapSchema;
